js/widgets/rhythmruler.js

```diff
@@ -1145,18 +1145,22 @@ function RhythmRuler () {
         // FIXME: rough workaround for #508, investigate reasons why
         // the backspace press doesn't work by default
         var numberInput = docById('dissectNumber');
-        numberInput.addEventListener('keydown', function (event) {
-            if (event.keyCode === BACKSPACE) {
-               numberInput.value = numberInput.value.substring(0, numberInput.value.length - 1);
-            }
-
+        numberInput.onkeydown =  function (event) {
+            if (event.keyCode == 46) {
+                numberInput.value = numberInput.value.substring(0, numberInput.value.length - 1);
+            }   
+        };
+        numberInput.oninput = function (event) {
             // Put a limit on the size (2 <--> 128).
-            if (numberInput.value < 2) {
-                numberInput.value = 2;
-            } else if (numberInput.value > 128) {
+            numberInput.onmouseout = function(){
+                if (numberInput.value < 2) {
+                    numberInput.value = 2;
+                } 
+            }
+            if (numberInput.value > 128) {
                 numberInput.value = 128;
             }
-        });
+        }
```

Comment:
Please use a ; at the end of function blocks.

Reply:
Sure :)